src/ethz_snow/snowing.py

```diff
@@ -0,0 +1,2227 @@
+
+        Raises:
+            TypeError: If k is not a dict.
+            ValueError: If a aspecific key is missing in dict k.
+            NotImplementedError: If freezing configuration is not correctly specified.
+            NotImplementedError: If model dimensionality is not correctly specified.
+            TypeError: If opcond is not of type operatingConditions.
+
+        Examples:
+            >>> S = Snowing()
+            >>> S = Snowing(k=d, opcond=op)
+            >>> S = Snowing(k=d, opcond=op, temperature="spatial_1D", configuration="shelf", plotting=True)
+            >>> S = Snowing(k=d, opcond=op, temperature="spatial_1D", configuration="shelf", plotting=False))
+            >>> S = Snowing(k=d, opcond=op, temperature="spatial_1D", configuration="visf", plotting=True))
+            >>> S = Snowing(k=d, opcond=op, temperature="spatial_2D", configuration="jacket"))
+        """
+
+        if not isinstance(k, dict):
+            raise TypeError(f"Input k must be of type dict. Was given {type(k)}.")
+        elif not all([key in k.keys() for key in HEATFLOW_REQUIREDKEYS]):
+            raise ValueError(
+                (
+                    f"A required key was missing from dictionary k, specifically "
+                    + f"{set(HEATFLOW_REQUIREDKEYS) - set(k.keys())}."
+                )
+            )
+        else:
+            self.k = k
+
+        # for saving the results
+        self._domain = None
+        self._prop = None
+        self._time = None
+        self._shelf = None
+        self._temp = None
+        self._ice = None
+
+        # dictionary of results
+        self._stats = {}
+
+        # number of repetitions
+        self.Nrep = Nrep
+
+        # dictionary of results for multiple simulations
+        if self.Nrep > 1:
+            self._statsMultiple = {}
+
+        # path to the parameters file
+        self.configPath = configPath
+
+        # simulation progress
+        self._simulationStatus = 0
+
+        # model complexity
+        if temperature is not None:
+            self.temperature = temperature
+            self.const["temperature"] = temperature
+        else:
+            self.temperature = self.const["temperature"]
+
+        # freezing configuration (shelf, VISF, etc.)
+        if configuration is not None:
+            self.configuration = configuration
+            self.const["configuration"] = configuration
+        else:
+            self.configuration = self.const["configuration"]
+
+        # check configuration
+        if self.configuration not in ["shelf", "VISF", "jacket"]:
+            raise NotImplementedError(
+                (
+                    f'Configuration "{self.configuration}" '
+                    + "not correctly specified, use shelf, jacket or VISF."
+                )
+            )
+
+        # check complexity
+        if self.temperature not in ["homogeneous", "spatial_1D", "spatial_2D"]:
```

Comment:
This should be checked in constants.py, not here

Reply:
Shouldn't it be checked here since I allow model dimensionality and freezing configurations as input keys besides the yaml file?